File: skills/whatsapp-chatbot-fastapi-starter/assets/fastapi-template/app/meta_client.py
import httpx

from app.settings import SETTINGS


async def send_text_message(to: str, text: str) -> dict:
    url = (
        f"https://graph.facebook.com/{SETTINGS.meta_api_version}"
        f"/{SETTINGS.whatsapp_phone_number_id}/messages"
    )

    payload = {
        "messaging_product": "whatsapp",
        "to": to,
        "type": "text",
        "text": {"body": text},
    }

    headers = {
        "Authorization": f"Bearer {SETTINGS.whatsapp_access_token}",
        "Content-Type": "application/json",
    }

    async with httpx.AsyncClient(timeout=10.0) as client:
        response = await client.post(url, json=payload, headers=headers)
        response.raise_for_status()
        return response.json()

File: skills/whatsapp-chatbot-nodejs-starter/assets/nodejs-express-template/src/metaClient.js
const axios = require("axios");
const { config } = require("./config");

const metaHttp = axios.create({
  baseURL: `https://graph.facebook.com/${config.metaApiVersion}`,
  headers: {
    Authorization: `Bearer ${config.whatsappAccessToken}`,
    "Content-Type": "application/json"
  },
  timeout: 10000
});

async function sendTextMessage(to, text) {
  const payload = {
    messaging_product: "whatsapp",
    to,
    type: "text",
    text: { body: text }
  };

  const response = await metaHttp.post(
    `/${config.whatsappPhoneNumberId}/messages`,
    payload
  );

  return response.data;
}

module.exports = { sendTextMessage };
